patches/react-native-share-menu+6.0.0.patch

```diff
@@ -1,22 +1,74 @@
+diff --git a/node_modules/react-native-share-menu/android/build.gradle b/node_modules/react-native-share-menu/android/build.gradle
+index 9557fdb..26825b5 100644
+--- a/node_modules/react-native-share-menu/android/build.gradle
++++ b/node_modules/react-native-share-menu/android/build.gradle
+@@ -1,12 +1,22 @@
+ apply plugin: 'com.android.library'
+ 
++buildscript {
++    ext {
++        rnsmDefaultTargetSdkVersion = 31
++        rnsmDefaultCompileSdkVersion = 31
++        rnsmDefaultMinSdkVersion = 21
++    }
++    ext.safeExtGet = {prop, fallback ->
++        rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
++    }
++}
++
+ android {
+-    compileSdkVersion 29
+-    buildToolsVersion "29.0.2"
++    compileSdkVersion safeExtGet('compileSdkVersion', rnsmDefaultCompileSdkVersion)
+ 
+     defaultConfig {
+-        minSdkVersion 16
+-        targetSdkVersion 29
++        minSdkVersion safeExtGet('minSdkVersion', rnsmDefaultCompileSdkVersion)
++        targetSdkVersion safeExtGet('targetSdkVersion', rnsmDefaultCompileSdkVersion)
+         versionCode 1
+         versionName "1.0"
+         ndk {
+diff --git a/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java b/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
+index 09abd7b..058dd1c 100644
+--- a/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
++++ b/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
+@@ -112,6 +112,16 @@ public class ShareMenuModule extends ReactContextBaseJavaModule implements Activ
+     clearSharedText();
+   }
+ 
++  @ReactMethod
++  public void addListener(String eventName) {
++    // Required for RN built in Event Emitter Calls.
++  }
++
++  @ReactMethod
++  public void removeListeners(Integer count) {
++    // Required for RN built in Event Emitter Calls.
++  }
++
+   private void dispatchEvent(ReadableMap shared) {
+     if (mReactContext == null || !mReactContext.hasActiveCatalystInstance()) {
+       return;
 diff --git a/node_modules/react-native-share-menu/ios/Constants.swift b/node_modules/react-native-share-menu/ios/Constants.swift
-index 08385b7..e22bdfa 100644
+index 08385b7..31279fd 100644
 --- a/node_modules/react-native-share-menu/ios/Constants.swift
 +++ b/node_modules/react-native-share-menu/ios/Constants.swift
 @@ -32,6 +32,7 @@ public let USER_DEFAULTS_EXTRA_DATA_KEY = "ShareMenuUserDefaultsExtraData"
  public let URL_SCHEME_INFO_PLIST_KEY = "AppURLScheme"
  public let HOST_APP_IDENTIFIER_INFO_PLIST_KEY = "HostAppBundleIdentifier"
  public let HOST_URL_SCHEME_INFO_PLIST_KEY = "HostAppURLScheme"
 +public let HOST_APP_GROUP_INFO_PLIST_KEY:String = "ReactNativeShareMenuAppGroup"
```

Comment:
Let's remove this typing, the other keys don't use it

Reply:
```suggestion
+public let HOST_APP_GROUP_INFO_PLIST_KEY = "ReactNativeShareMenuAppGroup"
```